ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/BasicAction.java

```diff
@@ -1498,7 +1498,11 @@ protected synchronized int End (boolean reportHeuristics)
             }
             else
             {
-                if (prepare(reportHeuristics) == TwoPhaseOutcome.PREPARE_NOTOK) {
+                int prepareStatus = prepare(reportHeuristics);
+
+                if (prepareStatus == TwoPhaseOutcome.PREPARE_NOTOK
+                        || (prepareStatus == TwoPhaseOutcome.PREPARE_ONE_PHASE_COMMITTED
```

Comment:
I think I should be changing the prepare instead to return TwoPhaseOutcome.ONE_PHASE_ERROR instead of PREPARE_ONE_PHASE_COMMITTED if the onePhaseCommit aborts.